src/Kotlin/DocumentationBuilder.kt

```diff
@@ -104,6 +125,16 @@ class DocumentationBuilder(val session: ResolveSession, val options: Documentati
             node.appendProjection(typeArgument)
     }
 
+    fun DocumentationNode.appendAnnotations(annotated: Annotated) {
+        annotated.getAnnotations().forEach {
+            val annotationNode = it.build()
+            if (annotationNode != null) {
+                append(annotationNode,
+                        if (annotationNode.name == "deprecated") DocumentationReference.Kind.Deprecation else DocumentationReference.Kind.Annotation)
```

Comment:
I'd rather not over-generalize too early. We can refactor this later when the requirements are more clear.